src/main/java/supertracker/parser/Parser.java

```diff
@@ -230,6 +240,20 @@ private static Command parseNewCommand(String input) throws TrackerException {
             throw new TrackerException(ErrorMessage.EMPTY_PARAM_INPUT);
         }
 
+        String dateString = null;
+        LocalDate expiryDate = LocalDate.parse(INVALID_EX_DATE, DateTimeFormatter.ofPattern(INVALID_EX_DATE_FORMAT));
+
+        boolean hasExpiry = !matcher.group(EX_DATE_GROUP).isEmpty();
+
+        try {
+            if (hasExpiry) {
+                dateString = matcher.group(EX_DATE_GROUP).trim().substring(2);
```

Comment:
I suggest using the replace method instead of the substring method.
```suggestion
                dateString = matcher.group(EX_DATE_GROUP).trim().replace(EX_DATE_FLAG + BASE_FLAG, "");
```